### PR TITLE
fix: Rate limiting hello

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

- Link to issue1 here
- Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/utils/RateLimiter.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class RateLimiter {
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/utils/RateLimiter.java:22]: needs to follow the same singleton pattern using resource distributor

File: src/main/java/io/supertokens/webserver/api/core/HelloAPI.java [hunk 2]
```diff
@@ -75,8 +76,14 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
 
     private void handleRequest(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         // API is app specific
-        // TODO add rate limiter to this API based on appIdentifier
+
         try {
+            RateLimiter rateLimiter = RateLimiter.getInstance(getAppIdentifierWithStorage(req), 200);
+            if (!rateLimiter.checkRequest()) {
+                super.sendTextResponse(429, "Too many requests", resp);
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/core/HelloAPI.java:83]:
```suggestion
     super.sendTextResponse(200, "Hello", resp);
```

File: src/main/java/io/supertokens/webserver/api/core/HelloAPI.java [hunk 3]
```diff
@@ -75,8 +76,14 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
 
     private void handleRequest(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         // API is app specific
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/core/HelloAPI.java:78]: add the rate limiter to `/` endpoint as well

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/utils/RateLimiter.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/utils/RateLimiter.java]: Done

File: src/main/java/io/supertokens/webserver/api/core/HelloAPI.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/core/HelloAPI.java]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**